/*
 * This file was automatically generated by EvoSuite
 * Sat Dec 24 09:29:01 GMT 2016
 */

package tutorial;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.NoSuchElementException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import tutorial.LinkedListIterator;
import tutorial.Node;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class LinkedListIterator_ESTest extends LinkedListIterator_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Node<Object> node0 = new Node<Object>((Object) null);
      LinkedListIterator<Object> linkedListIterator0 = new LinkedListIterator<Object>(node0);
      boolean boolean0 = linkedListIterator0.hasNext();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Node<Integer> node0 = new Node<Integer>((Integer) null);
      LinkedListIterator<Integer> linkedListIterator0 = new LinkedListIterator<Integer>(node0);
      linkedListIterator0.next();
      // Undeclared exception!
      try { 
        linkedListIterator0.next();
        fail("Expecting exception: NoSuchElementException");
      
      } catch(NoSuchElementException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("tutorial.LinkedListIterator", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      LinkedListIterator<Object> linkedListIterator0 = new LinkedListIterator<Object>((Node<Object>) null);
      boolean boolean0 = linkedListIterator0.hasNext();
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      LinkedListIterator<Object> linkedListIterator0 = new LinkedListIterator<Object>((Node<Object>) null);
      // Undeclared exception!
      try { 
        linkedListIterator0.remove();
        fail("Expecting exception: UnsupportedOperationException");
      
      } catch(UnsupportedOperationException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("tutorial.LinkedListIterator", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      Integer integer0 = new Integer(0);
      Node<Integer> node0 = new Node<Integer>(integer0);
      LinkedListIterator<Integer> linkedListIterator0 = new LinkedListIterator<Integer>(node0);
      Integer integer1 = linkedListIterator0.next();
      assertEquals(0, (int)integer1);
  }
}
